### PR TITLE
Feature: Use Hilt for DI + Fixed tests

### PR DESCRIPTION
**Pull request by ozonePowered**
Just added Hilt for dependency injection.
Documentation: https://developer.android.com/training/dependency-injection/hilt-android

Edit: also includes fixing tests and and a bit of refactoring

### CODE REVIEW

File: domain/build.gradle [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
 apply plugin: 'kotlin'
+apply plugin: 'kotlin-kapt'
 
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
 
+    implementation "com.google.dagger:hilt-android:$hilt_version"
+    kapt "com.google.dagger:hilt-android-compiler:$hilt_version"
+
```
Comment (ccreusot) [inline domain/build.gradle:11]: Our domain doesn't need this dependency

**Review by ccreusot — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/cedriccreusot/pokedex/PokemonModule.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+@Module
+@InstallIn(ActivityRetainedComponent::class)
+class PokemonModule {
+    @Provides
+    fun providePokeApi(): PokeApi = PokeApiClient()
+
+    @Provides
+    fun providePokemonRepository(
+        pokemonDataSource: PokeApi
+    ): PokemonRepository = PokemonRepositoryAdapter(pokemonDataSource)
```
Comment (ccreusot) [inline app/src/main/java/fr/cedriccreusot/pokedex/PokemonModule.kt:22]: If I follow the documentation you can use @Binds to give the interface to use.

File: app/src/main/java/fr/cedriccreusot/pokedex/dataadapter/PokemonRepositoryAdapter.kt [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import fr.cedriccreusot.domain.common.model.EmptyError
-import fr.cedriccreusot.domain.list.model.Pokemon
 import fr.cedriccreusot.domain.common.model.Result
 import fr.cedriccreusot.domain.common.model.Success
+import fr.cedriccreusot.domain.list.model.Pokemon
+import fr.cedriccreusot.domain.list.repository.PokemonRepository
 import me.sargunvohra.lib.pokekotlin.client.PokeApi
 
 class PokemonRepositoryAdapter(private val pokemonDataSource: PokeApi) : PokemonRepository {
```
Comment (ccreusot) [inline app/src/main/java/fr/cedriccreusot/pokedex/dataadapter/PokemonRepositoryAdapter.kt:8]: You could use `@Inject constructor(private val pokemonDataSource: PokeApi)`

File: app/src/main/java/fr/cedriccreusot/pokedex/PokemonModule.kt (hunk 2 above)
Comment (ozonePowered) [reply app/src/main/java/fr/cedriccreusot/pokedex/PokemonModule.kt:22]: True

**Comment by ozonePowered** [comment]
I just fixed tests, refactored some mocks in them.
I also found some broken imports due to classes without package definition.

File: domain/src/test/kotlin/fr/cedriccreusot/domain/list/usecase/FetchPokemonListUseCaseTest.kt [hunk 4]
```diff
@@ -1,54 +1,48 @@
...
-        every { repository.getPokemons() }.returns(
+        every { repository.getPokemons(any(), any()) }.returns(
             Success(
                 listOf(
-                    Pokemon(0, "Mew"),
-                    Pokemon(1, "Bulbizar"),
-                    Pokemon(2, "Sylvain")
+                    Pokemon(0, "Mew", "", "Psy", null),
+                    Pokemon(1, "Bulbizarre", "", "Grass", "Poison"),
+                    Pokemon(2, "Sylvain", "", "Developer", "ADSL")
```
Comment (Alezco) [inline domain/src/test/kotlin/fr/cedriccreusot/domain/list/usecase/FetchPokemonListUseCaseTest.kt:26]: 😂

**Review by Alezco — APPROVED** [review]

**Review by ccreusot — APPROVED** [review]

File: domain/src/test/kotlin/fr/cedriccreusot/domain/list/usecase/FetchPokemonListUseCaseTest.kt (hunk 4 above)
Comment (sylvain-tagheuer) [reply domain/src/test/kotlin/fr/cedriccreusot/domain/list/usecase/FetchPokemonListUseCaseTest.kt:26]: c'est ça moquez vous mais j'aurai bientôt la fibre et vous rigolerez moins :p

**Review by sylvain-tagheuer — APPROVED** [review]

**Pull request merged**